### PR TITLE
allow explicit tokenizer & text_encoder in unload_textual_inversion

### PR DESCRIPTION
**Pull request by H3zi**
# What does this PR do?

Fixes [#6974](https://github.com/huggingface/diffusers/issues/6974)

Currently, `unload_textual_inversion` only performs unload to tokenizer & text_encoder, originally designed for SD pipelines but doesn't support SDXL, which also uses tokenizer_2 & text_encoder_2.

By allowing passing explicitly the tokenizer & text_encoder to unload textual inversion modules (just like it allows in the `load_textual_inversion` function) we can support fully unload in SDXL pipelines as well.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sayakpaul @linoytsaban

### CODE REVIEW

**Comment by H3zi** [comment]
@sayakpaul Ive added a snippet based on the example demonstrated [here](https://huggingface.co/blog/sdxl_lora_advanced_script#inference) by @linoytsaban, I hope it's OK.

**Comment by yiyixuxu** [comment]
cc @fabiorigano can you take a look here?

File: src/diffusers/loaders/textual_inversion.py [hunk 1]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
 
         # Remove just one token
         pipeline.unload_textual_inversion("<moe-bius>")
+
+        # Examples 3 (based on SDXL LoRA Advanced training)
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:487]: here I would write "Example 3: unload from SDXL"
WDYT?

File: src/diffusers/loaders/textual_inversion.py [hunk 2]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
...
+        pipeline = AutoPipelineForText2Image.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0")
+        embedding_path = hf_hub_download(repo_id="LinoyTsaban/web_y2k_lora", filename="web_y2k_emb.safetensors", repo_type="model")
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:489]:
```suggestion
        embedding_path = hf_hub_download(repo_id="linoyts/web_y2k", filename="web_y2k_emb.safetensors", repo_type="model")
```

I couldn't find that repo_id, I tested this one `linoyts/web_y2k` and it works  @linoytsaban

File: src/diffusers/loaders/textual_inversion.py [hunk 3]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
...
+
+        # load embeddings to the text encoders
+        state_dict = load_file(embedding_path)
+
+        # load embeddings of text_encoder 1 (CLIP ViT-L/14)
+        pipeline.load_textual_inversion(state_dict["clip_l"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:495]:
```suggestion
        pipeline.load_textual_inversion(state_dict["clip_l"], token=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder, tokenizer=pipeline.tokenizer)
```

File: src/diffusers/loaders/textual_inversion.py [hunk 4]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
...
+        # load embeddings of text_encoder 2 (CLIP ViT-G/14)
+        pipeline.load_textual_inversion(state_dict["clip_g"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder_2, tokenizer=pipe.tokenizer_2)
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:497]:
```suggestion
        pipeline.load_textual_inversion(state_dict["clip_g"], token=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder_2, tokenizer=pipeline.tokenizer_2)
```

File: src/diffusers/loaders/textual_inversion.py [hunk 5]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
...
+
+        # Unload explicitly from both text encoders abd tokenizers
+        pipeline.unload_textual_inversion(tokens=["<s0>", "<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:500]:
```suggestion
        pipeline.unload_textual_inversion(tokens=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder, tokenizer=pipeline.tokenizer)
```

File: src/diffusers/loaders/textual_inversion.py [hunk 6]
```diff
@@ -481,11 +483,28 @@ def unload_textual_inversion(
...
+        pipeline.unload_textual_inversion(tokens=["<s0>", "<s1>"], text_encoder=pipe.text_encoder_2, tokenizer=pipe.tokenizer_2)
```
Comment (fabiorigano) [inline src/diffusers/loaders/textual_inversion.py:501]:
```suggestion
        pipeline.unload_textual_inversion(tokens=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder_2, tokenizer=pipeline.tokenizer_2)
```

**Review by fabiorigano — COMMENTED** [review]
thanks @yiyixuxu for adding me here

look good to me. I left a few nits

**Review by fabiorigano — APPROVED** [review]
LGTM!

**Pull request merged**